addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/protocol/NikoHomeControlCommunication.java

```diff
@@ -0,0 +1,878 @@
+
+    /**
+     * Return currency from Niko Home Control Controller, retrieved when connecting.
+     *
+     * @return currency
+     */
+    public String getCurrency() {
+        return currency;
+    }
+
+    private void setCurrency(String currency) {
+        this.currency = currency;
+    }
+
+    /**
+     * Return units of measure from Niko Home Control Controller, retrieved when connecting.
+     *
+     * @return units
+     */
+    public String getUnits() {
+        return units;
+    }
+
+    private void setUnits(String units) {
+        this.units = units;
+    }
+
+    /**
+     * Return timezone from Niko Home Control Controller, retrieved when connecting.
+     *
+     * @return timezone
+     */
+    public String getTz() {
+        return tz;
+    }
+
+    private void setTz(String tz) {
+        this.tz = tz;
+    }
+
+    /**
+     * Return daylight saving time from Niko Home Control Controller, retrieved when connecting.
+     *
+     * @return dst
+     */
+    public String getDst() {
+        return dst;
+    }
+
+    private void setDst(String dst) {
+        this.dst = dst;
+    }
+
+    /**
+     * Return last energy data erase time from Niko Home Control Controller, retrieved when connecting.
+     *
+     * @return lastenergyerase
+     */
+    public String getLastenergyerase() {
+        return lastenergyerase;
+    }
+
+    private void setLastenergyerase(String lastenergyerase) {
+        this.lastenergyerase = lastenergyerase;
+    }
+
+    /**
+     * Return last configuration upload time from Niko Home Control Controller, retrieved when connecting.
+     *
+     * @return lastconfig
+     */
+    public String getLastconfig() {
+        return lastconfig;
+    }
+
+    private void setLastconfig(String lastconfig) {
+        this.lastconfig = lastconfig;
+    }
+
+    /**
+     * Return the location name identified by locationID
+     *
+     * @param locationID
+     * @return location name
+     */
+    public String getLocationName(int locationID) {
+        return this.locationNames.get(locationID);
+    }
+
+    /**
+     * Return the list of all action IDs in the Niko Home Control Controller.
+     *
+     * @return <code>ArrayList&ltInteger></code> of action IDs
+     */
+    public ArrayList<Integer> getActions() {
+        return this.actions;
+    }
+
+    /**
+     * Get name of action identified by actionID.
+     *
+     * @param actionID
+     * @return action name
+     */
+    public String getActionName(int actionID) {
+        if (this.actionNames.containsKey(actionID)) {
+            return this.actionNames.get(actionID);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Get type of action identified by actionID.
+     * <p>
+     * Action type is 1 for a switch and 2 for a dimmer.
+     *
+     * @param actionID
+     * @return action type
+     */
+    public Integer getActionType(int actionID) {
+        if (this.actionNames.containsKey(actionID)) {
```

Comment:
I didn't matter. All of these would be there if it existed, but as in your previous suggestion, this code is not needed anymore.